clients/cli/tests/test_sync_metadata_to_token2022.rs

```diff
@@ -141,3 +252,322 @@ async fn test_sync_metadata_from_spl_token_to_token2022() {
+    const POLL_EVERY: Duration = Duration::from_millis(500);
+    const TIMEOUT: Duration = Duration::from_secs(10);
+
+    let start_time = Instant::now();
+    let deadline = start_time.checked_add(TIMEOUT).unwrap();
+
+    while Instant::now() < deadline {
+        if let Ok(resp) = env
+            .rpc_client
+            .get_account_with_config(wrapped_mint_address, RpcAccountInfoConfig::default())
+            .await
+        {
+            if resp
+                .value
+                .as_ref()
+                .is_some_and(|a| a.lamports >= expected_balance)
+            {
+                break; // state visible, proceed
+            }
+        }
+        sleep(POLL_EVERY).await;
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn test_sync_from_token2022_with_external_metaplex_pointer() {
+    let env = setup_test_env().await;
+
+    // 1. Create an external Metaplex metadata account
+    let dummy_mint = create_unwrapped_mint(&env, &spl_token::id()).await;
+    let name = "External Metaplex".to_string();
+    let symbol = "EXTM".to_string();
+    let uri = "http://external.com".to_string();
+    let metaplex_pda = create_metaplex_metadata(
+        &env,
+        &dummy_mint,
+        spl_token::id(),
+        name.clone(),
+        symbol.clone(),
+        uri.clone(),
+    )
+    .await;
+
+    // 2. Create unwrapped T22 mint pointing to the external metadata
+    let unwrapped_mint_kp = create_token2022_mint(
+        &env,
+        Some(metaplex_pda),
+        true,  // has_pointer_extension
+        false, // has_token_metadata
+    )
+    .await;
+    let unwrapped_mint = unwrapped_mint_kp.pubkey();
+
+    // 3. Create and fund wrapped mint
+    execute_create_mint(&env, &unwrapped_mint, &spl_token_2022::id()).await;
+    let wrapped_mint_address = get_wrapped_mint_address(&unwrapped_mint, &spl_token_2022::id());
+
+    let initial_balance = env
+        .rpc_client
+        .get_balance(&wrapped_mint_address)
+        .await
+        .unwrap();
+    let funding_amount = 1_000_000_000;
+    let fund_tx = Transaction::new_signed_with_payer(
+        &[transfer(
+            &env.payer.pubkey(),
+            &wrapped_mint_address,
+            funding_amount,
+        )],
+        Some(&env.payer.pubkey()),
+        &[&env.payer],
+        env.rpc_client.get_latest_blockhash().await.unwrap(),
+    );
+    env.rpc_client
+        .send_and_confirm_transaction(&fund_tx)
+        .await
+        .unwrap();
+
+    // The test validator can sometimes have a race condition for this specific test
+    // where the state of an account is not immediately reflected in the bank
+    // used for the next transaction's simulation. For that reason, we'll wait when
+    // the balance is reflected in an RPC query.
+    let expected_balance = initial_balance.saturating_add(funding_amount);
+    wait_until_ready(&env, &wrapped_mint_address, expected_balance).await;
```

Comment:
That's really strange... the rpc client coming out of test-validator uses `Processed` commitment, and that gets correctly propagated down to the preflight commitment for all `send_transaction` calls. So there shouldn't be any waiting required. Does it always error at this part?

It's also particularly strange that it only affects this test, since the other tests are all doing the funding 🤔